### PR TITLE
Block/card upload document

### LINKED ISSUE #78: [Request] card-upload-document
https://dribbble.com/shots/14028875/attachments/5645737?mode=media

image

### PR DESCRIPTION
**Pull request by bazsup**
closes #78 

Hi @siriwatknp,
⚠️ I have added the `react-dropzone` library for dragging files to the card. Please take a look.

Screenshot 2023-10-29 at 15 16 02

### CODE REVIEW

**Comment by siriwatknp** [comment]
| Results |  |
| --- | --- |
| Build Results | https://www.chromatic.com/build?appId=64e95d8f1f548945b1225cb8&number=74 |
| Storybook Preview | https://64e95d8f1f548945b1225cb8-yjwcwirlwy.chromatic.com/ |
| Component Count | 56 |

**Comment by siriwatknp** [comment]
@all-contributors please add @bazsup for code.

File: blocks/card-upload-document/src/CardUploadDocument.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import storyDialog from "../../../.storybook/decorators/storyDialog";
+import Usage from "../usage.mdx";
+import CardUploadDocument from "./CardUploadDocument";
+
+const meta = {
+  title: "Card/Upload/Document",
+  component: CardUploadDocument,
+  parameters: {
+    layout: "centered",
+    githubUsername: "", // (optional) Your github username. If provided, your avatar will be displayed in the story toolbar
```
Comment (siriwatknp) [inline blocks/card-upload-document/src/CardUploadDocument.stories.tsx:12]:
```suggestion
    githubUsername: "bazsup", // (optional) Your github username. If provided, your avatar will be displayed in the story toolbar
```

File: blocks/card-upload-document/src/CardUploadDocument.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }, []);
+  const { getRootProps, isDragActive } = useDropzone({
+    onDrop,
+    noClick: true,
+    noKeyboard: true,
+  });
+
+  return (
+    <div {...getRootProps()}>
+      <Card
```
Comment (siriwatknp) [inline blocks/card-upload-document/src/CardUploadDocument.tsx:23]:
```suggestion
      <Card
        {...getRootProps()}
```
It's better to spread the props directly to the card

File: blocks/card-upload-document/usage.mdx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Meta, Source } from "@storybook/blocks";
+import raw from "./src/CardUploadDocument?raw";
+
+<Meta title="Card/Upload/Document" />
```
Comment (siriwatknp) [inline blocks/card-upload-document/usage.mdx:4]:
```suggestion
<Meta title="Card/Upload/Document" />

## Dependencies

This block uses [react-dropzone](https://react-dropzone.js.org/) to implement the drag-and-drop experience.
```

File: blocks/card-upload-document/src/CardUploadDocument.tsx (hunk 2 above)
Comment (bazsup) [reply blocks/card-upload-document/src/CardUploadDocument.tsx:23]: I encountered a TypeScript error while attempting to spread the props into the Card component."

image

**Comment by siriwatknp** [comment]
| Results |  |
| --- | --- |
| Build Results | https://www.chromatic.com/build?appId=64e95d8f1f548945b1225cb8&number=75 |
| Storybook Preview | https://64e95d8f1f548945b1225cb8-ciorslxizj.chromatic.com/ |
| Component Count | 56 |

**Comment by siriwatknp** [comment]
| Results |  |
| --- | --- |
| Build Results | https://www.chromatic.com/build?appId=64e95d8f1f548945b1225cb8&number=76 |
| Storybook Preview | https://64e95d8f1f548945b1225cb8-uzyhhoyvjv.chromatic.com/ |
| Component Count | 56 |

File: blocks/card-upload-document/src/CardUploadDocument.tsx (hunk 2 above)
Comment (siriwatknp) [reply blocks/card-upload-document/src/CardUploadDocument.tsx]: Found a way to make TypeScript happy in https://github.com/siriwatknp/joy-treasury/pull/95/commits/eccf23bb6c5a5035d9a58819e881666b0e073d80:

```ts
<Card
  {...(getRootProps() as Omit<DropzoneRootProps, "color">)}
  variant="plain"
  sx={{ minWidth: 300, borderRadius: "xl", boxShadow: "lg" }}
  size="sm"
>
```

**Comment by siriwatknp** [comment]
| Results |  |
| --- | --- |
| Build Results | https://www.chromatic.com/build?appId=64e95d8f1f548945b1225cb8&number=77 |
| Storybook Preview | https://64e95d8f1f548945b1225cb8-zbwrptrvdp.chromatic.com/ |
| Component Count | 56 |

**Pull request merged**